File: Ch2LnkdLsts/isPldrmItrtv.cpp
#include <iostream>
#include "LinkedList.h"
using namespace std;

bool isPlnDrm(Node* head){
	bool flag = false;

	
	return flag;
}

int main(){

	LinkedList* ll1 = new LinkedList();
	LinkedList* ll2 = new LinkedList();

	//a palindrome
	ll1->addNode(7);
	ll1->addNode(8);
	ll1->addNode(3);
	ll1->addNode(9);
	ll1->addNode(3);
	ll1->addNode(8);
	ll1->addNode(7);

	//not a plndrm
	ll2->addNode(3);
	ll2->addNode(0);
	ll2->addNode(7);
	ll2->addNode(5);


	cout << "is list 1 a palindrome? " << isPlnDrm(ll1->head);
	//cout << "is list 2 a palindrome? " << isPlnDrm(ll2->head);

	return 0;
}

File: Ch2LnkdLsts/isPldrmItrtv.py
from LinkedList import LinkedList
import random


ll = LinkedList()
stack = []

def pop_stack():
	temp = stack[-1]
	del stack[-1]
	return temp

def isPalindrome():
	
	fast = ll.head
	slow = ll.head
	flag = True
	while (fast and fast.next):
		stack.append(slow.data)
		slow = slow.next
		fast = fast.next.next

	if fast:
		slow = slow.next

	while slow:
		if pop_stack() != slow.data:
			flag = False
			break
		slow = slow.next


	return flag

if __name__ == '__main__':
	
	ll.addNode(7)
	ll.addNode(2)
	ll.addNode(3)
	ll.addNode(2)
	ll.addNode(7)

	#for x in range(15):
	#	ll.addNode(random.randint(1,100))
	
	ll.printList()

	print "is the list a palindrome?",isPalindrome()

